api/v1alpha1/gateway_types.go

```diff
@@ -149,13 +165,20 @@ const (
 // - aws: https://docs.aws.amazon.com/elasticloadbalancing/latest/application/create-https-listener.html#describe-ssl-policies
 // - azure: https://docs.microsoft.com/en-us/azure/app-service/configure-ssl-bindings#enforce-tls-1112
 type ListenerTLS struct {
-	// Certificates is a list of certificates containing resources
-	// that are bound to the listener.
+	// Certificates is a reference to one or more Kubernetes objects each containing
+	// an identity certificate that is bound to the listener. The hostname in a TLS
+	// SNI client hello message is used for certificate matching and route hostname
+	// selection.
 	//
-	// If apiGroup and kind are empty, will default to Kubernetes Secrets resources.
+	// If unspecified, a wildcard certificate is generated for the subdomain of the
+	// listener's name.
```

Comment:
I think that wildcard certificate handling should be proposed and discussed as an independent item. Generating wildcard certificates is something that ought to require explicit configuration, and I don't think that it should happen as a side-effect of naming a listener.

I'll also note that many ingress controllers don't generate certificates, so I'm not sure the the spec should place requirements on them to do so.